pokemon-checker/src/repositories/PokemonRepository.ts

```diff
@@ -0,0 +1,98 @@
+import PokemonDTO from "../DataTransferObjects/PokemonDTO";
+import { Time } from "../constants/Time";
+import { PokemonFactory } from "../factories/PokemonFactory";
+
+export class PokemonRepository {
+  private pokemonTable: Record<string, PokemonDTO>;
+  private expiryTimestamp: number;
+  private factory: PokemonFactory
+
+  // Used to check against localStorage
+  private static storageTimestampKey = "expiryTimestamp";
+
+  constructor() {
+    this.pokemonTable = {};
+    this.factory = new PokemonFactory();
```

Comment:
I didn't like having a factory down here (normally the service handles that), but I needed a way to quickly convert back and forth between holding the full Pokemon data as JSON here vs. storing it as a string in `localStorage`.

Not sure what a permanent solution to this looks like, yet.